public/app/plugins/datasource/elasticsearch/datasource.js

```diff
@@ -211,38 +223,87 @@ function (angular, _, moment, kbn, ElasticQueryBuilder, IndexPattern, ElasticRes
 
     this.getFields = function(query) {
       return this._get('/_mapping').then(function(res) {
-        var fields = {};
+        var fields = {},
+            new_pre;
         var typeMap = {
           'float': 'number',
           'double': 'number',
           'integer': 'number',
           'long': 'number',
           'date': 'date',
           'string': 'string',
+          'nested': 'nested'
         };
 
-        for (var indexName in res) {
-          var index = res[indexName];
-          var mappings = index.mappings;
-          if (!mappings) { continue; }
-          for (var typeName in mappings) {
-            var properties = mappings[typeName].properties;
-            for (var field in properties) {
-              var prop = properties[field];
+        function autocompleteSearchFields(doc, pre) {
```

Comment:
ok looks like this was covered in https://github.com/grafana/grafana/pull/6043/files#diff-c80919dba20dd4b4f83090a6cc86c668R115